### DISCUSSION TITLE
How to pause consumers while doing maintenance and only process the exiting consumed message

### DISCUSSION THREAD

**Discussion by khanakia** [nats-io/nats-server#5263] [Q&A] [answered]
I have a use case not sure if others have this issue.

- I want to pause all the subscribers to stop listening for the new queues
- I want the subscribers to process already consumed queues
- Once all the existing consumed queues are finished and the subscriber will not accept any new queues I will stop the server
- Do maintenance and deploy the new binary
- Unpause and let the subscribers listen to the queue again

**I do not want to break the current process which is already consumed as my queues take hours to finish so if I stop the server in the middle all the current processes will break too.**

**Comment by ripienaar** [comment, marked as answer]
This will be in the next major release, see https://github.com/nats-io/nats-server/issues/4966
